src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -604,6 +604,27 @@ private void EmitCondBranchCore(BoundExpression condition, ref object dest, bool
                     EmitSequenceCondBranch(seq, ref dest, sense);
                     return;
 
+                case BoundKind.LoweredIsPatternExpression:
+                    var expression = (BoundLoweredIsPatternExpression)condition;
+                    dest ??= new object();
+
+                    if (sense)
+                    {
+                        _builder.MarkLabel(expression.WhenTrueLabel);
+                        _builder.EmitBranch(ILOpCode.Br, dest);
+                        _builder.MarkLabel(expression.WhenFalseLabel);
+                    }
+                    else
+                    {
+                        var doneLabel = new object();
+                        _builder.MarkLabel(expression.WhenTrueLabel);
+                        _builder.EmitBranch(ILOpCode.Br, doneLabel);
+                        _builder.MarkLabel(expression.WhenFalseLabel);
+                        _builder.EmitBranch(ILOpCode.Br, dest);
+                        _builder.MarkLabel(doneLabel);
+                    }
```

Comment:
Could we just do?
```
                        _builder.MarkLabel(expression.WhenFalseLabel);
                        _builder.EmitBranch(ILOpCode.Br, dest);
                        _builder.MarkLabel(expression.WhenTrueLabel);
```

Reply:
The fall-through is expected to be the true label.

Reply:
> The fall-through is expected to be the true label.

Are you saying that the ```WhenTrueLabel``` can be reached without an explicit jump? If so, that makes me even more uncomfortable about this new node

Reply:
It seems like leaf nodes are filtered out when we're checking for fall-through and those always get a label so this is a non-issue. I went with the original suggestion.

Also added handling of Sequence, LoweredIsPatternExpression in EmitCondExpr. Please take another look, thanks.